mlflow/tracking/client.py

```diff
@@ -1004,9 +1006,9 @@ def log_dict(self, run_id, dictionary, artifact_file):
             with open(tmp_path, "w") as f:
                 # TODO: Make `indent` and `sort_keys` configurable by exposing them as arguments
                 if extension in [".yml", ".yaml"]:
-                    yaml.dump(dictionary, f)
+                    yaml.dump(dictionary, f, indent=indent, sort_keys=sort_keys)
                 else:
-                    json.dump(dictionary, f)
+                    json.dump(dictionary, f, indent=indent, sort_keys=sort_keys)
```

Comment:
This is out of the scope of this PR but I think we should add `default_flow_style=False` to `yaml.dump`:

https://pyyaml.org/wiki/PyYAMLDocumentation#dictionaries-without-nested-collections-are-not-dumped-correctly

Reply:
```
document = """
  a: 1
  b:
    c: 3
    d: 4
"""
>>> print yaml.dump(yaml.load(document))
a: 1
b: {c: 3, d: 4}

>>> print yaml.dump(yaml.load(document), default_flow_style=False)
a: 1
b:
  c: 3
  d: 4
```